ukis_pysat/raster.py

```diff
@@ -117,6 +128,13 @@ def _pad_to_bbox(self, bbox, mode="constant", constant_values=0):
 
         self.__arr = destination
 
+        return self.__update_dataset()
+
+    def __update_dataset(self):
+        """Update dataset without writing to file after it theoretically changed.
+
+        :return: closed dataset in memory
+        """
         mem_profile = self.dataset.meta
         mem_profile.update(

```

Comment:
Does this update the whole dateset with all its attributes (e.g. also dataset.bounds)? The adjusted tests pass successfully so I assume yes.

Reply:
Yes, we are basically writing to a GDAL in-memory file, see [`In-Memory Files`](https://rasterio.readthedocs.io/en/latest/topics/memory-files.html#in-memory-files).